### DISCUSSION TITLE
How to run /robot_state node?

### DISCUSSION THREAD

**Discussion by CFBerryhill** [iRobotEducation/create3_docs#601] [Questions] [answered]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

pi 4 mounted

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.3

### Which RMW is your robot running?

I don't know

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

I want to launch the /robot_state node that I can see through node list as I enter the pi connected to the robot:

`
$ ros2 node list 
/_internal/composite_hazard
/_internal/kinematics_engine
/_internal/mobility
/_internal/stasis
/mobility_monitor
/motion_control
/robot_state
/static_transform
/system_health
/ui_mgr
`

I'm sourced and I'm built - but I can't figure out how to actually run this node. I thought I found the package online irobot_create_common so I tried 

`$ ros2 run irobot_create_common /robot_state`

but the package couldn't be found. I'm guessing that I'm just wrong on what the package is actually called.

still very new to ROS2 - I assume I'm missing something small / doing something silly. if someone I'm just not realizing that the node IS running, it isn't publishing to /odom at NEARLY 20 hz as [the documentation here](https://iroboteducation.github.io/create3_docs/api/odometry/) says. Any help appreciated.

**Comment by alsora** [comment, marked as answer]
If you see the node with `ros2 node list` it means that the node is running on the robot.
You should not try to run it yourself.
Users of create 3 can't run additional nodes on the robot

What you found online is part of the Create 3 Simulator and you should no try to use it in combination with a physical robot (it will cause problems as you will have multiple instances of the same node).

So, ignoring the first part, your actual problem is this one:

> it isn't publishing to /odom at NEARLY 20 hz as [the documentation here](https://iroboteducation.github.io/create3_docs/api/odometry/) says

Can you provide more details on this?
What commands are you running? Are you getting any odometry messages at all ? Are you getting data on the other topics?

**Reply by CFBerryhill** [reply]
aaaand I'm 2/2 for silly mistakes. Here was my odom callback

`def get_odom(self, msg):
        if self.odom is None:
             self.odom = msg
             self.pose_recieved = True
             self.get_logger().info('odom recieved: "%s"' % msg.header)`
             
So, that makes sense! Fixed now and recieving messages at a much more sensible rate.
